var/spack/repos/builtin/packages/cloverleaf/package.py

```diff
@@ -0,0 +1,102 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+import glob
+
+from spack import *
+
+
+class Cloverleaf(MakefilePackage):
+    """Proxy Application. CloverLeaf is a miniapp that solves the
+       compressible Euler equations on a Cartesian grid,
+       using an explicit, second-order accurate method.
+    """
+
+    homepage = "http://uk-mac.github.io/CloverLeaf"
+    url      = "http://mantevo.org/downloads/releaseTarballs/miniapps/CloverLeaf/CloverLeaf-1.1.tar.gz"
+
+    version('1.1', '65652b30a64eb237ec844a6fdd4cd518')
+
+    variant('build', default='ref', description='Type of Parallelism Build',
+            values=('cuda', 'mpi', 'openacc_cray', 'openmp', 'ref', 'serial'))
+
+    depends_on('mpi')
```

Comment:
Looked into it and they are using the mpi wrapper in all builds to abstract it.  Only the 'cuda', 'mpi', 'openacc_cray' and 'ref' builds depend on mpi.  I am making the changes.  Is there a more concise way of expressing this besides one at a time?
```
    depends_on('mpi', when='build=cuda')
    depends_on('mpi', when='build=mpi_only')
    depends_on('mpi', when='build=openacc_cray')
    depends_on('mpi', when='build=ref')
```